### ISSUE TITLE
CSS for standard names tables HTML pages

### ISSUE THREAD

**Issue by sadielbartholomew** [cf-convention/cf-convention.github.io#130]
I notice that all HTML pages for standard names tables are rendered in the browser without any styling, despite the rest of the CF Conventions site being styled with Bootstrap. Is there a reason that no styling has been applied to the HTML tables? If not, I think it would be better to apply some consistent (i.e. Bootstrap or some light customisation of) styling to the tables, for cross-site consistency in presentation if nothing else but probably making them a little easier to read and search through as well.

Unless anyone conveys of any reason to not add styling, I can put in a pull request to implement it quickly, as I have already tried out the built tables locally with some CSS. with the Bootstrap stylesheet used elsewhere in the site. I decided it would be better to use it with another very small stylesheet to customise the table elements themselves as they were otherwise cramped (see details below).

*************

## Demo of a HTML table page with (suggested) styling applied

*All screenshots shown here are taking the HTML version of the table for v.71 as an example, [as linked here](https://cfconventions.org/Data/cf-standard-names/71/build/cf-standard-name-table.html) for ease of comparison to the current (absence of) style (in the web sense).  I opened the pages locally in Firefox (68.0.1, 64-bit) and applied these sheets via the developer console, but of course they can (if I get the go-ahead) be coded into the headers of the HTML so they are applied permanently to the rendered pages on the site.*

### With just the Bootstrap stylesheet

There is already a Bootstrap CSS sheet applied to pages on the site, namely `Data/media/css/bootstrap.min.css`, so I started by trying that out on the page to see how it looked. While it generally looks clean and clear:

![css-for-sn-table-001](https://user-images.githubusercontent.com/30274190/93070250-8a97cf00-f676-11ea-88af-dddbe585c683.png)

the table itself has become cramped vertically, and cells have lost padding against the borders, as Bootstrap must amend the element & border spacing or similar:

![css-for-sn-table-003](https://user-images.githubusercontent.com/30274190/93070248-89ff3880-f676-11ea-8faa-2e30ca699980.png)

### Suggested style: Bootstrap stylesheet + small stylesheet to customise

To get over the above, I played around and found the following CSS customisations:

```css
td {
  height: 10px;
  line-height: 30px;
}

table td {
    padding-left: 8px;
    padding-right: 8px;
}

table th {
    padding-left: 8px;
    padding-right: 8px;
}

tr:hover {
  background-color: #f5f5f5;
}
```

would keep the consistent Bootstrap look and feel but stop the table itself (the most important part, of course!) being styled in a way that lacks spacing & looks cluttered.

With both the Bootstrap and the above stylesheets, we get the following (including greying of the row being hovered on to help match up names and units columns, etc.), which I propose as a candidate for the overall CSS to apply to the tables HTML pages:

![css-for-sn-table-sadesdone-002](https://user-images.githubusercontent.com/30274190/93070243-88ce0b80-f676-11ea-8504-032150c50799.png)

![css-for-sn-table-sadesdone001](https://user-images.githubusercontent.com/30274190/93070244-8966a200-f676-11ea-9820-61b0c5bf26f5.png)

Of course, people might prefer something a little different (e.g. a [striped table](https://www.w3schools.com/css/tryit.asp?filename=trycss_table_striped), maybe?) & perhaps there could some further styling "bells & whistles" added.

**Comment by erget** [comment]
@sadielbartholomew I can't think of a good reason not to use CSS (of course that doesn't mean that there isn't one!). It might be daunting to get this integrated with however they're built and deployed though - you might want to coordinate with Alison, as she probably knows of any pitfalls. My preference is for striped tables, beyond that I'm a known table slob, willing to consume the ugliest table, as long as it is indeed tabular. I'd support this initiative.

**Comment by davidhassell** [comment]
I don't know about the technicalities of sytle sheets, but I certainly support bringing the standard name table in line with the look and feel of the rest of the CF web site.

**Comment by sadielbartholomew** [comment]
Thanks @erget, I'll definitely aim to coordinate with Alison. I did suspect some deployment work might be required, since all the HTML table page outputs were in the `build` directory of each version so somehow the line(s) to specify to use certain stylesheets would have to get processed into all of those. Happy to do that though. Hopefully Alison can help with that, so I will get in touch with her to ask. Thanks for the pointer.

> My preference is for striped tables, beyond that I'm a known table slob, willing to consume the ugliest table, as long as it is indeed tabular.

:smile: Fair enough, I am sure there are many who don't care one bit as long as long as the tables are readable! But I do think unstyled HTML has the potential to look quite dated to some. It is certainly uncommon to see it on the general web nowadays (at least, outside of the personal webpages of academics!) If this gets to PR stage I will try out a striped table and I can show a screenshot so we can decide if we think that looks & reads better.

**Comment by sadielbartholomew** [comment]
Closed by #404.
